src/Test/Utilities/Portable/Metadata/ILValidation.cs

```diff
@@ -3,16 +3,175 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.IO;
 using System.Linq;
+using System.Reflection;
 using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Security.Cryptography;
 using System.Text;
+using System.Threading;
 using System.Xml;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis;
 using Microsoft.Metadata.Tools;
 
 namespace Roslyn.Test.Utilities
 {
     public static class ILValidation
     {
+        public static bool IsStreamSigned(Stream moduleContents)
+        {
+            moduleContents.Position = 0;
+
+            using (var metadata = ModuleMetadata.CreateFromStream(moduleContents))
+            {
+                var metadataReader = metadata.MetadataReader;
+                var peReader = metadata.Module.PEReaderOpt;
+                var peHeaders = peReader.PEHeaders;
+                var flags = peHeaders.CorHeader.Flags;
+
+                bool is32bit = peHeaders.PEHeader.Magic == PEMagic.PE32;
+                const int SectionHeaderSize = 40;
+                int peHeadersSize = peHeaders.PEHeaderStartOffset
+                    + PEHeaderSize(is32bit)
+                    + SectionHeaderSize * peHeaders.SectionHeaders.Length;
+
+                if (CorFlags.StrongNameSigned != (flags & CorFlags.StrongNameSigned))
+                {
+                    return false;
+                }
+
+                var snDirectory = peReader.PEHeaders.CorHeader.StrongNameSignatureDirectory;
+                int rva = snDirectory.RelativeVirtualAddress;
+                int size = snDirectory.Size;
+                var signature = peReader.GetSectionData(rva).GetContent(0, size);
+                if (!peHeaders.TryGetDirectoryOffset(snDirectory, out var snOffset))
+                {
+                    return false;
+                }
+
+                moduleContents.Position = 0;
+                int peSize = (int)moduleContents.Length;
```

Comment:
Should we use `checked` here and return false if there is an overflow? Not sure about the semantics of this field hence not sure how to interpret overflow here as a reviewer.

Reply:
I thought we had changed our defaults to checked everywhere?

Reply:
Our entire build is `unchecked`

https://github.com/dotnet/roslyn/blob/master/build/Targets/Settings.props#L316